lib/msal-core/src/ScopeSet.ts

```diff
@@ -130,13 +120,46 @@ export class ScopeSet {
      * Appends extraScopesToConsent if passed
      * @param {@link AuthenticationParameters}
      */
-    static appendScopes(reqScopes: Array<string>, reqExtraScopesToConsent: Array<string>): Array<string> {
-        if(reqScopes) {
-            return reqExtraScopesToConsent ? [...reqScopes, ...reqExtraScopesToConsent]: reqScopes;
+    static appendScopes(scopes: Array<string>, scopesToAppend: Array<string>): Array<string> {
+        if(scopes) {
+            return scopesToAppend ? [...scopes, ...scopesToAppend]: scopes;
         }
         return null;
     }
 
     // #endregion
 
+    /**
+     * append the required scopes: https://openid.net/specs/openid-connect-basic-1_0.html#Scopes
+     * @param scopes
+     */
+    static generateLoginScopes(scopes: Array<string>, clientId: string): Array<string> {
+        const loginScopes = [...scopes];
+        const clientIdIndex: number = loginScopes.indexOf(clientId);
+        if (this.isLoginScopes(loginScopes, clientId)) {
+            if (clientIdIndex >= 0) {
+                loginScopes.splice(clientIdIndex, 1);
+            }
+            if (loginScopes.indexOf("openid") === -1) {
+                loginScopes.push("openid");
```

Comment:
Does "openid" and "profile" exist as constants somewhere? I think they should, given how often they are referenced.

Reply:
Yes, the constants where already in place, now using them in generateLoginScopes.